website/docs/components/code-editor/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,104 @@
+### When not to use
+
+- As a read-only reference for code, use the [Code Block](/components/code-block) instead.
+
+## Standalone
+
+The `isStandalone` property increases the portability of the Code Editor to ensure that it can be used in different contexts. For example, a common use case of the Code Editor is in a "standalone" context, which can be part of a form, multi-step process, and is generally a part of the normal layout flow.
+
+![Code editor with rounded corners for enabled standalone property](/assets/components/code-editor/code-editor-rounded-standalone.png)
+
+Sometimes it may be necessary to use the Code Editor in a more dense layout or nested within another component.
+
+![Code editor embedded inside of a larger UI, where it is highlighted where the user is actively editing.](/assets/components/code-editor/code-editor-block-level.png)
+
+## Header
+
+The header is optionally available within the Code Editor. It consists of three sections that can be toggled as needed: the text content that includes the title and description, an actions container for secondary actions, and a custom yielded section of primary actions.
+
+### When to use a title and description
+
+A title and description provides additional contextual information for the Code Editor. An accessible name is mandatory and can either be provided by the title or by an external text element.
+
+!!! Do
+
+This provides an accessible name for the Code Editor so that a user and assistive technology can understand its purpose.
+
+![A partial form with a radio group labeled with "Enforement behavior" and then the options "Advisory" with helper text "failed policies produce a warning", "Soft mandatory" with helper text "Failed policies can be overriden", "Hard mandatory" with helper text "Failed policies stop the run". Then the Code Editor with an external acessible name/title "Policy code (Sentinel)"](/assets/components/code-editor/code-editor-external-do-accessible-name.png)
+
+!!!
+
+!!! Dont
+
+
+Assuming a user would understand the intent of the Code Editor without providing an accessible name will cause confusion and fail on accessibility requirements.
+![A partial form with a radio group labeled with "Enforement behavior" and then the options "Advisory" with helper text "failed policies produce a warning", "Soft mandatory" with helper text "Failed policies can be overriden", "Hard mandatory" with helper text "Failed policies stop the run". The Code Editor has no accessible name.](/assets/components/code-editor/code-editor-dont-external-accessible-name.png)
+
+!!!
+
+## Secondary actions
+
+The secondary actions section is for two optional buttons: [Copy Button](/components/copy/button) and a button to expand the editor. The Copy Button copies the content of the Code Editor to the clipboard. The other button maximizes and minimizes the Code Editor from inline to full screen. These actions are considered supporting functions less vital to the user’s work but may be useful.
```

Comment:
This is a bit more active voice, also might be a bit more clear to say exactly what the second button does (fill the viewport, full screen, etc). I'm not totally sold on this suggestion though, so feel free to completely ignore me

```suggestion
The secondary actions section supports two optional buttons: [Copy Button](/components/copy/button) and a button to expand the editor to fill the viewport. The Copy Button copies the content of the Code Editor to the clipboard, while the second button maximizes and minimizes the Code Editor from inline to full screen. These actions are considered supporting functions less vital to the user’s work but may be useful.
```